### ISSUE TITLE
The finetuned model appears to require a .pt file. How can I download or generate one?

### ISSUE THREAD

**Issue by Pelhans** [allenai/OLMo#450]
### ❓ The question

I have two GPUs, and when I use the load_path option, I get an error saying rank0.pt not found".

**Comment by epwalsh** [comment]
@Pelhans what checkpoints are you trying to load from?

**Comment by Pelhans** [comment]
1 B model, for the next command line
![image](https://github.com/allenai/OLMo/assets/17537939/337bebb5-d402-48ce-a19e-8b08622ec757)

**Comment by Pelhans** [comment]
I downloaded the PyTorch model in .bin format and generated a .pt file from the .bin file. I then set the model to restore in unshared mode, but it still seems not to work....
![image](https://github.com/allenai/OLMo/assets/17537939/9fa86411-a004-4d56-ad86-3b75f141c12f)
![image](https://github.com/allenai/OLMo/assets/17537939/b40916d1-3ff8-44d9-a7b4-8bd677668800)
![image](https://github.com/allenai/OLMo/assets/17537939/19b59ecc-340b-4bea-8d01-14dbe79ed482)

**Comment by Pelhans** [comment]
it seems that checkpoint.py can only load sharded .pt model.....I try to load the model which I train from zero. sharded model can load correct, but the unsharded model I got the same error with upper img...So is those param.object only exist when sharded model？ 
![image](https://github.com/allenai/OLMo/assets/17537939/3321f6a0-8237-491b-a5ee-51e7cbb19641)

**Comment by epwalsh** [comment]
You can load an unsharded checkpoint as long as the training script recognizes it as an unsharded checkpoint. And to ensure that, just give the checkpoint directory a name that ends with `-unsharded`.

**Comment by 0novanta** [comment]
@Pelhans how did you mange to load a checkpoint?
It seems that doing:

`torch.save(olmo.state_dict(), "./checkpoints/model.pt")`

is not enough.

**Comment by Pelhans** [comment]
> @Pelhans how did you mange to load a checkpoint? It seems that doing:
> 
> `torch.save(olmo.state_dict(), "./checkpoints/model.pt")`
> 
> is not enough.

Yes, It seems need a .pt file generated from FSDP....I finally  finetune this model from HuggingFace script...

**Comment by 0novanta** [comment]
@Pelhans could you copy here the code/steps that you followed?

**Comment by Pelhans** [comment]
> @Pelhans could you copy here the code/steps that you followed?
you can use lower scripy or use LLaMA-Factory directly..... this script is not a good way to finetune( large gpu usage and slowly...)

import hf_olmo
from datasets import load_dataset
from transformers import AutoModelForCausalLM, AutoTokenizer
from transformers import DataCollatorForLanguageModeling
from transformers import TrainingArguments, Trainer
import os
os.environ["WANDB_DISABLED"] = "true"

# Initialize process group.
#import torch.distributed as dist
#dist.init_process_group(backend="nccl")

olmo = AutoModelForCausalLM.from_pretrained("../olmo-main/ori_checkpoint/")
tokenizer = AutoTokenizer.from_pretrained("../olmo-main/ori_checkpoint/")

olmo = olmo.to('cuda')
dataset = load_dataset("../olmo-main/final_to_train_data/sum_out")
test_dataset = load_dataset("../olmo-main/final_to_train_data/test_data")
def tokenize_function(examples):
    return tokenizer(examples["text"], padding="max_length", truncation=True, max_length=512)

tokenized_datasets = dataset.map(tokenize_function, batched=True)
test_tokenized_datasets = test_dataset.map(tokenize_function, batched=False)
small_train_dataset = tokenized_datasets["train"]
test_small_train_dataset = test_tokenized_datasets["train"].shuffle(seed=42).select(range(100))

data_collator = DataCollatorForLanguageModeling(tokenizer=tokenizer, mlm=False)

training_args = TrainingArguments(output_dir="./out_dir/", 
    evaluation_strategy="epoch",
    overwrite_output_dir=True,
    prediction_loss_only=True,
    per_device_train_batch_size=4,
    per_device_eval_batch_size=4,
    save_steps=500,
    gradient_accumulation_steps=16,
    save_total_limit=5,
    num_train_epochs=1,
    learning_rate=3e-5,
    adam_beta2=0.95,
    adam_epsilon=1e-5,
    weight_decay=0.1,
    fp16=True,
    fsdp=["shard_grad_op"],
    fsdp_config={"activation_checkpointing":True},
    warmup_ratio=0.05,

)

trainer = Trainer(
    model=olmo,
    args=training_args,
    train_dataset=small_train_dataset,
    eval_dataset=test_small_train_dataset,
    data_collator=data_collator
)

trainer.train()
trainer.save_model("./my_train")
